managed/resource_user.go

```diff
@@ -0,0 +1,296 @@
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"user_id": {
+				Description: "The ID of the user. Created automatically when a user is invited. Use this ID to get a specific user.",
+				Type:        types.StringType,
+				Computed:    true,
+				Optional:    true,
+			},
+			"email": {
+				Description: "The email of the user.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"role_name": {
+				Description: "The role of the user.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"user_name": {
+				Description: "The name of the user.",
+				Type:        types.StringType,
+				Computed:    true,
+				Optional:    true,
+			},
+			"user_state": {
+				Description: "The state of the user.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+		},
+	}, nil
+}
+
+func (r resourceUserType) NewResource(_ context.Context, p tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	return resourceUser{
+		p: *(p.(*provider)),
+	}, nil
+}
+
+type resourceUser struct {
+	p provider
+}
+
+func getUserPlan(ctx context.Context, plan tfsdk.Plan, user *User) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	diags.Append(plan.GetAttribute(ctx, path.Root("user_id"), &user.UserID)...)
+	diags.Append(plan.GetAttribute(ctx, path.Root("email"), &user.Email)...)
+	diags.Append(plan.GetAttribute(ctx, path.Root("role_name"), &user.RoleName)...)
+
+	return diags
+}
+
+// Invite user
+func (r resourceUser) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider wasn't configured before being applied, likely because it depends on an unknown value from another resource.",
+		)
+		return
+	}
+
+	var plan User
+	var accountId, message string
+	var getAccountOK bool
+	resp.Diagnostics.Append(getUserPlan(ctx, req.Plan, &plan)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Debug(ctx, "Error while getting the plan for the user")
+		return
+	}
+
+	apiClient := r.p.client
+
+	accountId, getAccountOK, message = getAccountId(ctx, apiClient)
+	if !getAccountOK {
+		resp.Diagnostics.AddError("Unable to get account ID", message)
+		return
+	}
+
+	if (!plan.UserID.Unknown && !plan.UserID.Null) || plan.UserID.Value != "" {
+		resp.Diagnostics.AddError(
+			"User ID provided for new user",
+			"The user_id was provided even though a new user is being invited. Do not include this field in the provider when inviting a user.",
+		)
+		return
+	}
+
+	projectId, getProjectOK, message := getProjectId(ctx, apiClient, accountId)
+	if !getProjectOK {
+		resp.Diagnostics.AddError("Unable to get project ID ", message)
+		return
+	}
+
+	email := plan.Email.Value
+	roleName := plan.RoleName.Value
+
+	roleResp, response, err := apiClient.RoleApi.ListRbacRoles(ctx, accountId).RoleTypes("ALL").Limit(100).DisplayName(roleName).Execute()
```

Comment:
How would the user know what are good values for the role field? Can you please add some better documentation?

Reply:
Yeah i was not sure of this- that how indeed would the TF users know of the available roles. And by documentation do you mean the description of the role name field? Still confused about what would be helpful there in this regard..